nexus/src/external_api/http_entrypoints.rs

```diff
@@ -6428,34 +6430,107 @@ impl NexusExternalApi for NexusExternalApiImpl {
 
     async fn login_saml_begin(
         rqctx: RequestContext<Self::Context>,
-        path_params: Path<params::LoginToProviderPathParam>,
-        query_params: Query<params::LoginUrlQuery>,
+        _path_params: Path<params::LoginToProviderPathParam>,
+        _query_params: Query<params::LoginUrlQuery>,
     ) -> Result<Response<Body>, HttpError> {
-        console_api::login_saml_begin(rqctx, path_params, query_params).await
+        let apictx = rqctx.context();
+        let handler = async { console_api::serve_console_index(&rqctx).await };
+        apictx
+            .context
+            .external_latencies
+            .instrument_dropshot_handler(&rqctx, handler)
+            .await
     }
 
     async fn login_saml_redirect(
         rqctx: RequestContext<Self::Context>,
         path_params: Path<params::LoginToProviderPathParam>,
         query_params: Query<params::LoginUrlQuery>,
     ) -> Result<HttpResponseFound, HttpError> {
-        console_api::login_saml_redirect(rqctx, path_params, query_params).await
```

Comment:
Some of these are slightly more interesting than inlining a function call. This helper function did all the stuff you normally see here at top level in this file (instrumentation call, pulling stuff out of path and query params, etc), _and_ the nitty gritty datastore logic that normally goes inside `nexus/src/app`. So I pulled the former out to here and the latter into `nexus/src/app/login.rs`.